src/journal/journald-syslog.c

```diff
@@ -333,7 +336,9 @@ void server_process_syslog_message(
                 return;
 
         if (s->forward_to_syslog)
-                forward_syslog_raw(s, priority, orig, ucred, tv);
+                forward_syslog_raw(s, priority, (char *) orig, buf_len, ucred, tv);
+
+        buf = strstrip((char *) buf);
 
```

Comment:
so, we got the argument passed in as "const", and here you modify the string then. That's not OK. Parameters that are const may not be modified...

Now, simply dropping the const in the parameter here I don't like either, it makes for weird calling conventions... I mean it's usually a good thing to operate on buffers that aren't modified. hence maybe do it like this:

 ```c
size_t n;
for (n = buf_len; n > 0; n --)
        if (!strchr(WHITESPACE, e[n-1]))
                break;
buf = strndupa(buf, e - n); 
```